packages/commerce-sdk-react/src/hooks/useAsync.ts

```diff
@@ -24,21 +25,27 @@ export const useAsync = <T>(fn: () => Promise<T>, deps?: unknown[]): QueryRespon
     return result
 }
 
-export const useAsyncExecute = <A, R>(fn: (arg: A) => Promise<R>) => {
+export const useAsyncExecute = <A extends any[], R>(fn: (...args: A) => Promise<R>) => {
```

Comment:
Aside: have we considered the name `useAsyncCallback`? Other libraries have used that name. And I think `callback` is more clear than `execute`. With `callback`, there is a sense of time with it... that the function you pass in would be called later and not immediately.

Reply:
This name was deliberately chosen to be bad, so that it would have to be renamed!